test/cppsuite/src/component/component.cpp

```diff
@@ -47,21 +49,14 @@ void
 component::load()
 {
     logger::log_msg(LOG_INFO, "Loading component: " + _name);
-    _enabled = _config->get_optional_bool(ENABLED, true);
-    /* If we're not enabled we shouldn't be running. */
-    _running = _enabled;
-
-    if (!_enabled)
-        return;
-
-    _sleep_time_ms = _config->get_throttle_ms();
 }
 
 void
 component::run()
 {
     logger::log_msg(LOG_INFO, "Running component: " + _name);
-    while (_enabled && _running) {
+    _running = true;
```

Comment:
This is a smart change, removing `_enabled` from the conditional.

If it is easy to try to run a not-enabled component: I'd suggest you consider a runtime assertion.

Straight `boolean` will likely work but, I'd recommend you look at `std::atomic_flag` which I believe will be more predictable/reliable.

Can `do_work()` throw an exception? If it can is it expected to crash the system, or be handled at higher level? If not does it make sense to place the while loop in a `try` statement?

Reply:
No exceptions are thrown in the framework, that was a decision taken during the design.

Regarding the assertion, are you saying I should switch the flag to a value once the function has been executed once and make sure it is never executed again?